Composer/packages/lib/indexers/src/dialogUtils/validation.ts

```diff
@@ -29,8 +29,8 @@ const isExpression = (value: string | boolean | number, types: string[]): boolea
 //TODO: returnType is number, schem type is string, need map or unify
 const checkReturnType = (returnType: ReturnType, types: string[]): string => {
   return returnType === ReturnType.Object ||
-    ~types.indexOf(ExpressionTypeMapString[returnType]) ||
-    (returnType === ReturnType.Number && ~types.indexOf(ExpressionType.integer))
+    types.includes(ExpressionTypeMapString[returnType]) ||
+    (returnType === ReturnType.Number && types.includes(ExpressionType.integer))
```

Comment:
cant this just be types[ExpressionType.integer] instead of doing includes everytime?

Reply:
👍

Reply:
Looking at this again, I'm not sure we can do anything about it. Types is an array but ExpressionType.integer is a string; we want to check if the string is a value in the array, so `includes` is appropriate here. The more correct thing here would be to make types a `Set<string>`, but that ought to be its own cleanup ticket later - it's a non-trivial amount of work.